File: Code/OfRectanglesAndSquares/OfRectanglesAndSquares/Square.cs
namespace OfRectanglesAndSquares
{
    /// <summary>
    /// This class stores the status of a Square object
    /// TODO breaks Liskov
    /// </summary>
    public class Square : Rectangle
    {
        public override void SetWidth(double width)
        {
            base.SetWidth(width);
            base.SetHeight(width);
        }

        public override void SetHeight(double height)
        {
            base.SetHeight(height);
            base.SetWidth(height);
        }

        public Square(double side) : base(side, side)
        {
        }
    }
}

File: Code/OfRectanglesAndSquares.Java/src/OfRectanglesAndSquares/Square.java
package OfRectanglesAndSquares;

/**
 * This class stores the status of a Square object
 */
public class Square implements ISquare {

    private double _side;
    
    public void setSide(double side) {
        this._side = side;
    }
    
    public double getSide() {
		return _side;
	}

	public Square(double side) {
    	this._side = side;
    }
}
